jenkins/janitor.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import collections
+import datetime
+import json
+import re
+import subprocess
+import sys
+
+# A resource that need to be cleared.
+Resource = namedtuple('Resource', 'name condition status')
+demolish_order = [
+Resource("instances", "zone", None),
+Resource("addresses", "region", None),
+Resource("disks", "zone", None),
+Resource("firewall-rules", None, None),
+Resource("routes", None, None),
+Resource("forwarding-rules", "region", None),
+Resource("target-pools", "region", None),
+Resource("instance-groups", "zone", "managed"),
+Resource("instance-templates", None, None),
+# Beaware of insertion order
+]
+
+def Collect(project, age, resource, filt):
```

Comment:
`collect`

It would also be nice to document the arguments and return value. See, for example, the comments on `job_results` in [this file](https://github.com/kubernetes/test-infra/blob/master/jenkins/test-history/gen_html.py).